src/swap/bitcoin.rs

```diff
@@ -101,13 +101,7 @@ impl hbit::ExecuteRefund for Wallet {
 }
 
 impl Wallet {
-    pub async fn redeem(
```

Comment:
Interesting. This has uncovered a little problem with implementations of `ExecuteRefund`. Thank you! I'll open a PR for that soon.